### ISSUE TITLE
Not launching on Windows 11

### ISSUE THREAD

**Issue by pa-august** [bibendovsky/bstone#375]
I have what I believe is a bstone folder that was working on Windows 10. 

I pulled it over from an external drive to a laptop I have Windows 11 on and it doesn't seem to launch. Clicking on the exe just produces the hourglass mouse icon for a moment, then nothing happens.

Here's my bstone_log:

```
BStone v1.2.11

Looking for game content.
Found "Blake Stone: Aliens Of Gold" v2.1.
Content source: working dir

Data path: "C:\LocalGames\Blake Stone - Aliens of Gold\"
Mod path: ""
Profile path: "C:\Users\anton\AppData\Roaming\bibendovsky\bstone\"
Screenshot path: "C:\Users\anton\AppData\Roaming\bibendovsky\bstone\"

[SND] Initializing audio
[SND] ------------------
[SND_OAL] Default library: "OpenAL32.dll"
[SND_OAL] Library: ""
[SND_OAL] Device name: ""
[SND_OAL] Device names:
[SND_OAL] 	"Generic Hardware"
[SND_OAL] 	"Generic Software"
[SND_OAL] Default device name: "Generic Hardware"
[SND_OAL] Current device name: "Generic Software"
[SND_OAL] Device extensions:
[SND_OAL] 	ALC_ENUMERATION_EXT
[SND_OAL] 	ALC_EXT_CAPTURE
[SND_OAL] 	ALC_EXT_EFX
[SND_OAL] Version: 1.1
[SND_OAL] Renderer: Software
[SND_OAL] Vendor: Creative Labs Inc.
[SND_OAL] Context extensions:
[SND_OAL] 	EAX
[SND_OAL] 	EAX2.0
[SND_OAL] 	EAX3.0
[SND_OAL] 	EAX4.0
[SND_OAL] 	EAX5.0
[SND_OAL] 	EAX3.0EMULATED
[SND_OAL] 	EAX4.0EMULATED
[SND_OAL] 	AL_EXT_OFFSET
[SND_OAL] 	AL_EXT_LINEAR_DISTANCE
[SND_OAL] 	AL_EXT_EXPONENT_DISTANCE
```

I searched my C drive for `bstone_config` and didn't find anything.

**Comment by bibendovsky** [comment]
Try to run with these parameters:
`bstone.exe --vid_renderer software --vid_is_windowed 1 --vid_windowed_width 640 --vid_windowed_height 480 --snd_driver 2d_sdl`

**Comment by pa-august** [comment]
> Try to run with these parameters:
> `bstone.exe --vid_renderer software --vid_is_windowed 1 --vid_windowed_width 640 --vid_windowed_height 480 --snd_driver 2d_sdl`

That worked, and now works without the command line parameters as well.

**Comment by pa-august** [comment]
The problem seems to be with the non-2d_sdl sound driver, attempting to change the sound driver setting while running the game causes a ctd.

**Comment by bibendovsky** [comment]
Could you post crash information from Windows log?
`Event Viewer -> Windows Logs -> Application`

**Comment by pa-august** [comment]
> Could you post crash information from Windows log?
> `Event Viewer -> Windows Logs -> Application`

`Log Name:      Application
Source:        Application Error
Date:          8/1/2021 6:12:38 AM
Event ID:      1000
Task Category: (100)
Level:         Error
Keywords:      Classic
User:          N/A
Computer:      DESKTOP-12OHJ11
Description:
Faulting application name: bstone.exe, version: 1.2.11.0, time stamp: 0x60579f26
Faulting module name: wrap_oal.dll, version: 2.1.4.0, time stamp: 0x452c326d
Exception code: 0xc0000005
Fault offset: 0x00000000000340d5
Faulting process id: 0x293c
Faulting application start time: 0x01d786bdb240c92a
Faulting application path: C:\LocalGames\Blake Stone - Aliens of Gold\bstone.exe
Faulting module path: C:\WINDOWS\SYSTEM32\wrap_oal.dll
Report Id: 7c278233-5c43-4530-94b7-07511317af85
Faulting package full name: 
Faulting package-relative application ID: 
Event Xml:
<Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
  <System>
    <Provider Name="Application Error" />
    <EventID Qualifiers="0">1000</EventID>
    <Version>0</Version>
    <Level>2</Level>
    <Task>100</Task>
    <Opcode>0</Opcode>
    <Keywords>0x80000000000000</Keywords>
    <TimeCreated SystemTime="2021-08-01T10:12:38.6298762Z" />
    <EventRecordID>6545</EventRecordID>
    <Correlation />
    <Execution ProcessID="11164" ThreadID="0" />
    <Channel>Application</Channel>
    <Computer>DESKTOP-12OHJ11</Computer>
    <Security />
  </System>
  <EventData>
    <Data>bstone.exe</Data>
    <Data>1.2.11.0</Data>
    <Data>60579f26</Data>
    <Data>wrap_oal.dll</Data>
    <Data>2.1.4.0</Data>
    <Data>452c326d</Data>
    <Data>c0000005</Data>
    <Data>00000000000340d5</Data>
    <Data>293c</Data>
    <Data>01d786bdb240c92a</Data>
    <Data>C:\LocalGames\Blake Stone - Aliens of Gold\bstone.exe</Data>
    <Data>C:\WINDOWS\SYSTEM32\wrap_oal.dll</Data>
    <Data>7c278233-5c43-4530-94b7-07511317af85</Data>
    <Data>
    </Data>
    <Data>
    </Data>
  </EventData>
</Event>`

**Comment by bibendovsky** [comment]
[bstone_bin_win7_64_v1_2_12-20210801-f0a9687.zip](https://github.com/bibendovsky/bstone/files/6913214/bstone_bin_win7_64_v1_2_12-20210801-f0a9687.zip)

Please check how this build works with OpenAL.
(Built from f0a9687d6bd889e5199018961fe6d3c2aba54ce1)

**Comment by pa-august** [comment]
> [bstone_bin_win7_64_v1_2_12-20210801-f0a9687.zip](https://github.com/bibendovsky/bstone/files/6913214/bstone_bin_win7_64_v1_2_12-20210801-f0a9687.zip)
> 
> Please check how this build works with OpenAL.
> (Built from [f0a9687](https://github.com/bibendovsky/bstone/commit/f0a9687d6bd889e5199018961fe6d3c2aba54ce1))

Still crashing:

```Log Name:      Application
Source:        Application Error
Date:          8/1/2021 6:01:20 PM
Event ID:      1000
Task Category: (100)
Level:         Error
Keywords:      Classic
User:          N/A
Computer:      DESKTOP-12OHJ11
Description:
Faulting application name: bstone.exe, version: 1.2.11.0, time stamp: 0x60579f26
Faulting module name: wrap_oal.dll, version: 2.1.4.0, time stamp: 0x452c326d
Exception code: 0xc0000005
Fault offset: 0x00000000000340d5
Faulting process id: 0x2778
Faulting application start time: 0x01d78720b0dcf001
Faulting application path: C:\LocalGames\Blake Stone - Aliens of Gold\bstone.exe
Faulting module path: C:\WINDOWS\SYSTEM32\wrap_oal.dll
Report Id: f9624420-dfd6-4213-b14b-d857ed5baa70
Faulting package full name: 
Faulting package-relative application ID: 
```

**Comment by bibendovsky** [comment]
[bstone_bin_win7_64_v1_2_12-20210804-3f38b15.zip](https://github.com/bibendovsky/bstone/files/6934047/bstone_bin_win7_64_v1_2_12-20210804-3f38b15.zip)


Added debug messages for OpenAL calls (3f38b15d1d0f770fda0b0efc56ca1e63c8984712).
Please attach the log `bstone_log.txt` (preferably ZIPed) after the crash.

**Comment by pa-august** [comment]
[bstone_log.zip](https://github.com/bibendovsky/bstone/files/6938946/bstone_log.zip)

**Comment by bibendovsky** [comment]
[bstone_bin_win7_64_v1_2_12-20210806-04554f2.zip](https://github.com/bibendovsky/bstone/files/6946776/bstone_bin_win7_64_v1_2_12-20210806-04554f2.zip)

Added more debug messages and make log file reopen for each write (04554f2afafe278af10fc9cac29feec3bb666f5a).

Could you also provide OpenAL runtime (`oalinst.exe`) you used on Windows 11?

**Comment by pa-august** [comment]
[bstone_log_updated_openal.zip](https://github.com/bibendovsky/bstone/files/6949264/bstone_log_updated_openal.zip)
[bstone_log.zip](https://github.com/bibendovsky/bstone/files/6949265/bstone_log.zip)

My version of OpenAL probably came along with the Steam version of Puzzle Quest. I updated to 1.1 from openal.org and the game doesn't crash but does chug. bstone_log is from before I updated to 1.1 and updated if from after.

**Comment by bibendovsky** [comment]
> the game doesn't crash but does chug

It should be. Usually log updates on initialization. To find out where crash happens I have 1) to make sure log message will be written into log, and 2) to add checkpoint messages at OpenAL calls.

[bstone_bin_win64_v1_2_12-20210808-830c8ae-mingw-8.1.0.zip](https://github.com/bibendovsky/bstone/files/6950920/bstone_bin_win64_v1_2_12-20210808-830c8ae-mingw-8.1.0.zip)

Seems only the port built in MinGW-W64 environment works with OpenAL v1.9.9.
Other environments (Visual C++ v142 / v141_xp, MSYS2 GCC 10.3.0, MSYS2 Clang 12.0.1) - fails.

**Comment by pa-august** [comment]
That revision is indeed working correctly with OpenAL, let me know if there's any more logs you want.

**Comment by bibendovsky** [comment]
> 
> 
> That revision is indeed working correctly with OpenAL, let me know if there's any more logs you want.

Thanks. For now it's enough.
I made simple test program for OpenAL v1.9.9.
It also works only if built in MinGW-W64 environment.
So I'm afraid there is nothing can be done more on my side, besides to release only MinGW-W64 builds for Windows in the future.

[test_oal_1_9_9_0_src_only.zip](https://github.com/bibendovsky/bstone/files/6951027/test_oal_1_9_9_0_src_only.zip)

**Comment by bibendovsky** [comment]
Fixed in v1.2.12 release (#381).
